### DISCUSSION TITLE
Support positiv/negative lookahead/lookbehing regex syntax

### DISCUSSION THREAD

**Discussion by bastilimbach** [community/community#8653] [Other Feature Feedback, Questions, & Ideas] [answered]
It would be great if the regex engine would support lookahead & lookbehinds to support searches like `/init(?=\W)/` (Search the word `init` which ends with any non-word character). 

This is probably not supported due to the reason that the code search is build in rust and the rust regex crate doesn't support it: 
>This crate provides a library for parsing, compiling, and executing regular expressions. Its syntax is similar to Perl-style regular expressions, but **lacks a few features like look around and backreferences**. In exchange, all searches execute in linear time with respect to the size of the regular expression and search text.

https://docs.rs/regex/latest/regex/ 😢

**Comment by h-vetinari** [comment]
Another closely related feature that unfortunately does not seep supported by the regex crate: [possessive qualifiers](https://github.com/orgs/community/discussions/118651) (which go hand in hand with lookaheads to me, because if there's something optional before the lookahead, you definitely don't want the engine to backtrack)

**Comment by look** [comment, marked as answer]
Thanks for the feedback. As noted by the documentation you quoted, these regular expression features cause non-linear runtimes. This is why we do not support them and do not plan to support them.
